### ISSUE TITLE
text-index fails when behind authorisation: currently no workaround

### ISSUE THREAD

**Issue by alpapan** [GMOD/jbrowse-components#4201]
When behind www authentication, text-index cannot access URLs.

while we can **update** indexes using --file and something like this:
```
for i in "trackId1;uri1;assembly1" "trackId2;uri2;assembly2"; do
    IFS=";" read track_uid file_uri assembly <<< "$i"
    file_path=${file_uri/$CLOUD_BASEURL/$DATA_BASEDIR}
    echo "Assembly: $assembly, ID: $track_uid, File: $file_path"
    jbrowse text-index --out $JBROWSE_LOCAL --exclude=CDS,exon,intron,three_prime_cis_splice_site,five_prime_cis_splice_site,five_prime_UTR,three_prime_UTR --file="$file_path" --fileId="$track_uid" --assemblies="$assembly"
done
```

this `--file` approach does not create entries in the config.json (not sure why). 

Only way currently is to turn off server authentication, build the indexes once so the config.json is created and then turn auth on again.

would be good if the --file approach had the option of creating/updating config.json entries.

The alternative is to allow a username/password to text-index

**Comment by cmdcolin** [comment]
--file is designed for embedded/programmatic usages where there is no config.json. you can manually update the config.json after using --file potentially

but it is true, there are a number of scenarios like auth that the tool can't handle. another option would be to run the indexing "locally" and then "rsync it to the server with the auth turned on"

**Comment by alpapan** [comment]
I have been using --file as it's the only way to index files behind authentication.

How difficult would it be to allow the jbrowse tools to send a credential?

**Comment by cmdcolin** [comment]
I am not sure how difficult it would be. is it correct that your config.json is referring to "remote files that are behind http basic authentication"?

**Comment by alpapan** [comment]
yes, my json is referrring to files behind http basic authentication (in this case an ldap)
when i was playing around with puppeteer i noticed that there was a way to do it...

**Comment by cmdcolin** [comment]
the workaround would probably be

- manually download the auth file
- replace the config temporarily with that local referred file
- run text-index
- then revert back to remote auth reference
might close as not planned cause it's just a lot to add any auth logic to cli
